File: frontend/projects/admin-console/src/app/core/models/popia-document-summary.dto.ts
export interface PopiaDocumentSummaryDto {
  totalDocuments: number;
  documentsWithNoPersonalInfo: number;
  documentsWithPersonalInfo: number;
  documentsWithSpecialPersonalInfo: number;
  byCategory: PopiaDocumentSummaryByCategoryDto[];
  byDepartment: PopiaDocumentSummaryByDepartmentDto[];
  byOwner: PopiaDocumentSummaryByOwnerDto[];
}

export interface PopiaDocumentSummaryByCategoryDto {
  category: string;
  total: number;
  withNoPersonalInfo: number;
  withPersonalInfo: number;
  withSpecialPersonalInfo: number;
}

export interface PopiaDocumentSummaryByDepartmentDto {
  department: string;
  total: number;
  withNoPersonalInfo: number;
  withPersonalInfo: number;
  withSpecialPersonalInfo: number;
}

export interface PopiaDocumentSummaryByOwnerDto {
  ownerUserId: string;
  total: number;
  withNoPersonalInfo: number;
  withPersonalInfo: number;
  withSpecialPersonalInfo: number;
}

File: application/src/MaemoCompliance.Application/Documents/Dtos/PopiaDocumentSummaryDto.cs
using MaemoCompliance.Domain.Documents;

namespace MaemoCompliance.Application.Documents.Dtos;

public class PopiaDocumentSummaryDto
{
    public int TotalDocuments { get; set; }
    public int DocumentsWithNoPersonalInfo { get; set; }
    public int DocumentsWithPersonalInfo { get; set; }
    public int DocumentsWithSpecialPersonalInfo { get; set; }
    
    public List<PopiaDocumentSummaryByCategoryDto> ByCategory { get; set; } = new();
    public List<PopiaDocumentSummaryByDepartmentDto> ByDepartment { get; set; } = new();
    public List<PopiaDocumentSummaryByOwnerDto> ByOwner { get; set; } = new();
}

public class PopiaDocumentSummaryByCategoryDto
{
    public string Category { get; set; } = null!;
    public int Total { get; set; }
    public int WithNoPersonalInfo { get; set; }
    public int WithPersonalInfo { get; set; }
    public int WithSpecialPersonalInfo { get; set; }
}

public class PopiaDocumentSummaryByDepartmentDto
{
    public string Department { get; set; } = null!;
    public int Total { get; set; }
    public int WithNoPersonalInfo { get; set; }
    public int WithPersonalInfo { get; set; }
    public int WithSpecialPersonalInfo { get; set; }
}

public class PopiaDocumentSummaryByOwnerDto
{
    public string OwnerUserId { get; set; } = null!;
    public int Total { get; set; }
    public int WithNoPersonalInfo { get; set; }
    public int WithPersonalInfo { get; set; }
    public int WithSpecialPersonalInfo { get; set; }
}

